lib/SILGen/SILGenApply.cpp

```diff
@@ -3622,8 +3622,9 @@ SILGenFunction::tryEmitAddressableParameterAsAddress(ArgumentSource &&arg,
     auto vd = cast<VarDecl>(memberStorage);
     // TODO: Is it possible and/or useful for class storage to be
     // addressable?
-    if (!vd->getDeclContext()->getInnermostTypeContext()
-         ->getDeclaredTypeInContext()->getStructOrBoundGenericStruct()) {
+    if (!vd->isInstanceMember()
+        || !vd->getDeclContext()->getInnermostTypeContext()
+             ->getDeclaredTypeInContext()->getStructOrBoundGenericStruct()) {
```

Comment:
is getInnermostTypeContext() correct to check here at all? This is not an instance property for instance:

```
struct S {
  func f() {
    var x = ... // innermost type context of `x` is a struct; why does that matter at all?
  }
}
```

I think you can just do `isa<StructDecl>(vd->getDeclContext())`?